modules/javafx.graphics/src/main/resources/com/sun/prism/es2/glsl/main1Light.frag

```diff
@@ -82,8 +85,12 @@ void main()
     vec3 l = normalize(lightTangentSpacePositions[0].xyz);
     d = clamp(dot(n,l), 0.0, 1.0)*(lights[0].color).rgb;
     s = pow(clamp(dot(-refl, l), 0.0, 1.0), power)*lights[0].color.rgb;
+	
+	//FalcoTheBold - added attenuation calculation for a single light
+	float dist = (lights[0].pos.xyz - gl_Position) / lights[0].atten.range;
```

Comment:
I don't understand this. If the distance from the pixel to the light is `lights[0].pos.xyz - gl_Position`, why divide by the range? If that distance is larger than the range, the light should have no effect. I don't see how this happens.

Reply:
Honestly, tiredness got the better of me that time and i was led too much by findings in other forums regarding how to use the range in these places. Truly an error on my part there. There will be a conditional element regarding the affected range instead in the next commit as well.